src/mon/OSDMonitor.h

```diff
@@ -428,6 +428,9 @@ class OSDMonitor : public PaxosService {
 
   int load_metadata(int osd, map<string, string>& m, ostream *err);
   void count_metadata(const string& field, Formatter *f);
+public:
+  void count_metadata(const string& field, map<string,int> *out);
+protected:
```

Comment:
I'm pretty unhappy about the duplication of hte metadata code across all teh *Monitor classes but factoring it out is a non-trivial cleanup due to a few inconsistencies.  Kicking that can down the road...